src/EppoClient.php

```diff
@@ -143,27 +140,33 @@ function () use ($configLoader) {
             }
         );
 
-        self::$instance = self::createAndInitClient($configLoader, $poller, $assignmentLogger, $isGracefulMode);
+        self::$instance = self::createAndInitClient($configLoader, $poller, $assignmentLogger, $isGracefulMode, throwOnFailedInit: $throwOnFailedInit);
 
         return self::$instance;
     }
 
     /**
-     * @throws EppoClientInitializationException
+     * @throws EppoClientInitializationException|InvalidConfigurationException
      */
     private static function createAndInitClient(
         ConfigurationLoader $configLoader,
         PollerInterface $poller,
         ?LoggerInterface $assignmentLogger,
         ?bool $isGracefulMode,
-        ?IBanditEvaluator $banditEvaluator = null
+        ?IBanditEvaluator $banditEvaluator = null,
+        ?bool $throwOnFailedInit = false,
     ): EppoClient {
         try {
             $configLoader->reloadConfigurationIfExpired();
         } catch (HttpRequestException | InvalidApiKeyException $e) {
-            throw new EppoClientInitializationException(
-                'Unable to initialize Eppo Client: ' . $e->getMessage()
-            );
+            $message = 'Unable to initialize Eppo Client: ' . $e->getMessage();
+            if ($throwOnFailedInit) {
```

Comment:
I guess the existing behavior is that `isGracefulMode` is not applicable during the init, so `throwOnFailedInit` is the only one we need to check here

Reply:
that's correct. At least, that was my thinking here.
I know in some SDKs, we have the separate flag for throwing on init and for this reason, I split it here. I don't think it would be problematic to use `isGracefulMode` to cover both init and on `getAssignment`

Reply:
Cool. Yep what you did here makes sense -- keeping the original behavior as-is